internal/ingress/annotations/annotations.go

```diff
@@ -44,44 +47,100 @@ const (
 	hostHeaderKey        = "/host-header"
 	methodsKey           = "/methods"
 
+	// RequireClassHandling indicates that ingress.class must be present and match
+	RequireClassHandling = "required"
+	// IgnoreClassHandling indicates that ingress.class is not required and is ignored
+	IgnoreClassHandling = "ignored"
+	// LazyClassHandling indicates that ingress.class can be empty if using the default class
+	LazyClassHandling = "optional"
+
 	// DefaultIngressClass defines the default class used
 	// by Kong's ingress controller.
 	DefaultIngressClass = "kong"
 )
 
-func validIngress(ingressAnnotationValue, ingressClass string, skipClasslessIngress bool) bool {
+func validIngress(ingressAnnotationValue, ingressClass string, classHandling string) (bool, error) {
```

Comment:
I don't see the rationale for the complicated diff in this file, can you explain why is this so complicated?
The whole purpose of https://docs.google.com/document/d/1qMwurrFiOMoy2EK569y6qY-9cLGKJrkTfloqNyp_J6o/edit is to simplify this function and make it more maintainable.